src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -38,46 +51,53 @@ func NewStatefulSet(
 	}
 }
 
+// Ensure will manage kubernetes v1.StatefulSet for redpanda.vectorized.io custom resource
 func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 	var sts appsv1.StatefulSet
 
-	err := r.Get(ctx, r.Key(), &sts)
-	if err != nil && !errors.IsNotFound(err) {
-		return err
+	getSTSErr := r.Get(ctx, r.key(), &sts)
```

Comment:
just wondering - why did you decide to choose a different name here?

Reply:
The `err` is shadowed in line [line 64](https://github.com/vectorizedio/redpanda/pull/544/files#diff-11cd0daf9c52813e733b2526ffd440b85ff6e9a8f6a11d8454c02cbd129bcb60R64). In retrospective this isn't right approach. Will revert that.